doc/tutorials/12-constant_pH/12-constant_pH.ipynb

```diff
@@ -166,95 +297,53 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "num_samples = 100\n",
+    "# the reference data from Henderson-Hasselbalch equation\n",
+    "def ideal_alpha(pH, pK):\n",
+    "    return 1. / (1 + 10**(pK - pH))\n",
     "\n",
-    "pHs = np.linspace(1, 8, num=15)\n",
-    "degrees_of_dissociation = []\n",
-    "std_dev_degree_of_dissociation = []\n",
-    "histograms = []\n",
-    "histogram_edges = range(0,N0+1)\n",
+    "# empty lists as placeholders for collecting data\n",
+    "numAs = []; # number of A- species observed at each sample\n",
     "\n",
+    "#run a productive simulation and collect the data\n",
+    "print(\"Simulated pH values: \",pHs)\n",
     "for pH in pHs:\n",
-    "    print(\"pH is {:.1f}\".format(pH))\n",
+    "    print(\"Run pH {:.2f} ...\".format(pH))\n",
     "    RE.constant_pH = pH\n",
-    "    RE.reaction(4 * N0) # equilibration to the new pH\n",
-    "    numAs = []\n",
+    "    numAs_current = []; # temporary data storage for a given pH\n",
+    "    RE.reaction(20*N_acid + 1) # pre-equilibrate to the new pH value\n",
     "    for i in range(num_samples):\n",
-    "        RE.reaction(N0 + 1)\n",
-    "        numAs.append(system.number_of_particles(type=type_A))\n",
-    "    degrees_of_dissociation.append(np.mean(numAs) / N0)\n",
-    "    std_dev_degree_of_dissociation.append(np.std(numAs, ddof=1) / N0)\n",
-    "    print(\"occurred particle numbers of type A \", sorted(set(numAs)))\n",
-    "    histogram = np.bincount(np.array(numAs).astype(int), minlength=N0 + 1)\n",
-    "    histograms.append(histogram / float(len(numAs)))\n",
-    "    if(abs(pH - 1.0) < 1e-4):\n",
-    "        print(\"acceptance rate dissociation\", RE.get_acceptance_rate_reaction(0))\n",
-    "        print(\"acceptance rate association\", RE.get_acceptance_rate_reaction(1))"
+    "        if(np.random.random()<PROB_REACTION): \n",
+    "            RE.reaction(N_acid + 1) # should be at least one reaction attempt per particle\n",
+    "        else:\n",
+    "            system.integrator.run(steps=1000);\n",
```

Comment:
You should make using the integrator conditional on USE_WCA in order to speed up the ideal case:

```
        else:
            if USE_WCA==True:
                system.integrator.run(steps=1000);
```

Reply:
Yes, we can do it.